protocol/communities/manager.go

```diff
@@ -819,9 +840,63 @@ func (m *Manager) HandleCommunityRequestToJoin(signer *ecdsa.PublicKey, request
 		return nil, err
 	}
 
+	if community.AcceptRequestToJoinAutomatically() {
+		err = m.markRequestToJoin(signer, community)
+		if err != nil {
+			return nil, err
+		}
+		requestToJoin.State = RequestToJoinStateAccepted
+	}
+
 	return requestToJoin, nil
 }
 
+func (m *Manager) HandleCommunityRequestToJoinResponse(signer *ecdsa.PublicKey, request *protobuf.CommunityRequestToJoinResponse) error {
+
+	community, err := m.persistence.GetByID(m.identity, request.CommunityId)
+	if err != nil {
+		return err
+	}
+	if community == nil {
+		return ErrOrgNotFound
+	}
+
+	communityDescriptionBytes, err := proto.Marshal(request.Community)
+	if err != nil {
+		return err
+	}
+
+	// We need to wrap `request.Community` in an `ApplicationMetadataMessage`
+	// of type `CommunityDescription` because `UpdateCommunityDescription` expects this.
+	//
+	// This is merely for marsheling/unmarsheling, hence we attaching a `Signature`
+	// is not needed.
+	metadataMessage := &protobuf.ApplicationMetadataMessage{
+		Payload: communityDescriptionBytes,
+		Type:    protobuf.ApplicationMetadataMessage_COMMUNITY_DESCRIPTION,
+	}
```

Comment:
@Samyoul Any objections here to leave the signature out?

Reply:
This is fine